src/ResultAggregator.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace PhpWorkshop\PhpWorkshop;
+
+/**
+ * Class ResultAggregator
+ * @package PhpWorkshop\PhpWorkshop
+ * @author Aydin Hassan <aydin@hotmail.co.uk>
+ */
+class ResultAggregator
+{
+    /**
+     * @var Fail[]|Success[]
+     */
+    private $results = [];
+
+    /**
+     * @param Fail|Success $result
+     */
+    public function add($result)
```

Comment:
Don't like this, but I don't really care for now, if you can come up with a nicer solution, I'm all ears.